lib/chef/knife/core/bootstrap_context.rb

```diff
@@ -187,16 +187,32 @@ def knife_config
         end
 
         #
-        # chef version string to fetch the latest current version from omnitruck
-        # If user is on X.Y.Z, bootstrap will use the latest X release
+        # Determine that CLI arguments to retrieve the correct version of Chef Infra Client from omnitruck
+        # By default, bootstrap will look for the latest version of the currently-running major release of on stable.
+        #
+        # @return [String] CLI arguments to pass into the chef download helper script
         def latest_current_chef_version_string
-          chef_version_string = if knife_config[:bootstrap_version]
-                                  knife_config[:bootstrap_version]
-                                else
-                                  Chef::VERSION.split(".").first
-                                end
+          # NOTE: Changes here should also be reflected in Knife::Core::BootstrapContext#latest_current_chef_version_string
+          installer_version_string = []
+          use_current_channel = (@config[:channel] == "current")
+          installer_version_string << "-p" if use_current_channel
```

Comment:
`-p` is actually back-compat in `install.sh`, we should pass `-c CHANNEL` install the installer_version_string here.

It is easiest to see here: https://omnitruck.chef.io/install.sh

But it comes from here: https://github.com/chef/mixlib-install/blob/master/lib/mixlib/install/generator/bourne/scripts/script_cli_parameters.sh